cmd/client/txgen/main.go

```diff
@@ -86,141 +96,87 @@ func main() {
+	setting := Settings{
 		NumOfAddress:      10000,
-		CrossShard:        false, // len(shardIDs) > 1,
 		MaxNumTxsPerBatch: *maxNumTxsPerBatch,
-		CrossShardRatio:   *crossShardRatio,
 	}
 
 	// TODO(Richard): refactor this chuck to a single method
 	// Setup a logger to stdout and log file.
-	logFileName := fmt.Sprintf("./%v/txgen.log", *logFolder)
+	logFileName := fmt.Sprintf("./%v/txgen-%v.log", *logFolder, shardID)
 	h := log.MultiHandler(
 		log.StreamHandler(os.Stdout, log.TerminalFormat(false)),
 		log.Must.FileHandler(logFileName, log.LogfmtFormat()), // Log to file
 	)
 	log.Root().SetHandler(h)
 
 	// Nodes containing blockchain data to mirror the shards' data in the network
-	nodes := []*node.Node{}
+
 	host, err := p2pimpl.NewHost(&selfPeer, nodePriKey)
 	if err != nil {
 		panic("unable to new host in txgen")
 	}
-	for _, shardID := range shardIDs {
-		node := node.New(host, &consensus.Consensus{ShardID: shardID}, nil, false)
-		// Assign many fake addresses so we have enough address to play with at first
-		nodes = append(nodes, node)
-	}
-
-	// Client/txgenerator server node setup
-	consensusObj, err := consensus.New(host, 0, nil, p2p.Peer{}, nil)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error :%v \n", err)
-		os.Exit(1)
-	}
-
-	clientNode := node.New(host, consensusObj, nil, false)
-	clientNode.Client = client.NewClient(clientNode.GetHost(), shardIDs)
-
-	readySignal := make(chan uint32)
-
-	// This func is used to update the client's blockchain when new blocks are received from the leaders
-	updateBlocksFunc := func(blocks []*types.Block) {
-		utils.GetLogInstance().Info("[Txgen] Received new block", "block num", blocks[0].NumberU64())
-		for _, block := range blocks {
-			for _, node := range nodes {
-				shardID := block.ShardID()
-
-				if node.Consensus.ShardID == shardID {
-					// Add it to blockchain
-					utils.GetLogInstance().Info("Current Block", "block num", node.Blockchain().CurrentBlock().NumberU64())
-					utils.GetLogInstance().Info("Adding block from leader", "txNum", len(block.Transactions()), "shardID", shardID, "preHash", block.ParentHash().Hex())
-					node.AddNewBlock(block)
-					stateMutex.Lock()
-					node.Worker.UpdateCurrent()
-					stateMutex.Unlock()
-					readySignal <- shardID
-				} else {
-					continue
-				}
-			}
-		}
-	}
-	clientNode.Client.UpdateBlocks = updateBlocksFunc
-
-	clientNode.NodeConfig.SetRole(nodeconfig.ClientNode)
-	clientNode.NodeConfig.SetIsClient(true)
-	clientNode.ServiceManagerSetup()
-	clientNode.RunServices()
-
-	// Start the client server to listen to leader's message
-	go func() {
-		// wait for 3 seconds for client to send ping message to leader
-		// FIXME (leo) the readySignal should be set once we really sent ping message to leader
-		time.Sleep(3 * time.Second) // wait for nodes to be ready
-		for _, i := range shardIDs {
-			readySignal <- i
-		}
-	}()
+	node := node.New(host, &consensus.Consensus{ShardID: uint32(shardID)}, nil, true) //Make it archival node.
```

Comment:
don't use the variable name the same as the package name.  It is very confusing. clientNode is better, IMHO.